bin/cheribsdtest/cheribsdtest_vm.c

```diff
@@ -431,19 +431,22 @@ create_tempfile()
  */
 CHERIBSDTEST(cheribsdtest_vm_notag_tmpfile_shared,
     "check tags are not stored for tmpfile() MAP_SHARED pages",
-    .ct_flags = CT_FLAG_SIGNAL | CT_FLAG_SI_CODE | CT_FLAG_SI_TRAPNO,
+    .ct_flags = CT_FLAG_SIGNAL | CT_FLAG_SI_CODE | CT_FLAG_SI_TRAPNO | CT_FLAG_SI_ADDR,
     .ct_signum = SIGSEGV,
     .ct_si_code = SEGV_STORETAG,
     .ct_si_trapno = TRAPNO_STORE_CAP_PF,
     .ct_check_xfail = xfail_need_writable_tmp)
 {
+	void *mmap_ret;
 	void * __capability volatile *cp;
 	void * __capability cp_value;
 	int fd, v;
 
 	fd = create_tempfile();
-	cp = CHERIBSDTEST_CHECK_SYSCALL(mmap(NULL, getpagesize(),
+	mmap_ret = CHERIBSDTEST_CHECK_SYSCALL(mmap(NULL, getpagesize(),
 	    PROT_READ | PROT_WRITE, MAP_SHARED, fd, 0));
+	cp = mmap_ret;
+	cheribsdtest_set_expected_si_addr(mmap_ret);
```

Comment:
```suggestion
	cheribsdtest_set_expected_si_addr((void *)(uintptr_t)(ptraddr_t)mmap_ret);
```
is what will actually be in si_addr, so we should probably set it to that regardless of whether we're doing exact equality or not

Reply:
Related to above, do you think we should change the type of the argument to `expected_si_addr` to `ptraddr_t`?

Reply:
No, I think we want the expected value to match what is in siginfo_t and we should be explicit here about what we expect that to mean (see my reply in the earlier comment as well).

Reply:
si_addr is a `void *` and in the cases where it reports the program counter rather than the loaded from/stored to address it does (currently, whether or not that's a good idea is a separate question) give you the full capability, so in theory you might want to use this interface to check that. Though deriving the capability to give it is rather awkward, you'd have to write assembly and and have a label referencing a specific instruction, so not sure.

Reply:
Yes, I'd thought about that. There is a GNU extension to C (`&& label`) to get the address of a goto label. It appears to be supported by clang...

Reply:
I don't know that we are going to write tests testing PC values in `si_addr` anytime soon.  I suspect if we did we might make them fuzzy though and just require `si_addr` to be within the bounds of the function and call it good enough.  Regardless, storing the full `si_addr` value as the expected value is what we want I think.  For these tests we want to assert that `si_addr` is null-derived and for that we need to compare the full capability.

Reply:
`&&label` gets you the address of somewhere to jump to that will give you the semantics of the original code, but there's no guarantee as to what instructions are there (even if you use inline assembly there could be any number of instructions in between).